src/Config/ObjectModel/TelemetryOptions.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Config.ObjectModel;
+
+/// <summary>
+/// Represents the options for telemetry.
+/// </summary>
+public record TelemetryOptions(ApplicationInsightsOptions ApplicationInsights)
+{ }
```

Comment:
Yes If Telemetry option is provided, by default it is configured with default values of ApplicationInsightsOptions, which is disabled and empty connection string.

Reply:
making it nullable

Reply:
For the record, this is a shorthand for defining the property itself/constructor.
Did you mean to set the property nullable? or want to change it only for the constructor? 

If the latter, you need to define a separate constructor. See my PR for initializing optional properties for `RuntimeOptions` for an e.g.

Reply:
nvm, I realized if we dont keep the optional property itself nullable, I will modify my PR - it might be an issue with the merge configuration file feature as well: [Using two config files](https://learn.microsoft.com/en-us/azure/data-api-builder/data-api-builder-cli#using-data-api-builder-with-two-configuration-files)
If the overriding config has minimal values, it will now have the optional values set to their defaults as opposed to being null. These defaults may not be the same as the base config and this could lead to a situation where the merged configuration overrides the base values just because they were written to the file system by us in the overriding config even when the developer might not have provided those initially. 
 
I think the best way to handle this is to re-introduce optional properties to be nullable, and leave them as is – don’t write the expanded version on the file system - so we don’t accidentally merge and create an incorrect merged configuration.